pkg/capr/configserver/server.go

```diff
@@ -84,21 +90,36 @@ func New(clients *wrangler.Context) *RKE2ConfigServer {
 		machines:                 clients.CAPI.Machine(),
 		bootstrapCache:           clients.RKE.RKEBootstrap().Cache(),
 		provisioningClusterCache: clients.Provisioning.Cluster().Cache(),
+		provisioningClusters:     clients.Provisioning.Cluster(),
+		mgmtClusterCache:         clients.Mgmt.Cluster().Cache(),
+		mgmtClusters:             clients.Mgmt.Cluster(),
 		k8s:                      clients.K8s,
 	}
 }
 
 func (r *RKE2ConfigServer) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	if !r.secrets.Informer().HasSynced() || !r.clusterTokens.Informer().HasSynced() {
+	if !r.secrets.Informer().HasSynced() || !r.clusterTokens.Informer().HasSynced() || !r.provisioningClusters.Informer().HasSynced() || !r.mgmtClusters.Informer().HasSynced() {
 		if err := r.secrets.Informer().GetIndexer().Resync(); err != nil {
 			logrus.Errorf("error re-syncing secrets informer in rke2configserver: %v", err)
 		}
 		if err := r.clusterTokens.Informer().GetIndexer().Resync(); err != nil {
 			logrus.Errorf("error re-syncing clustertokens informer in rke2configserver: %v", err)
 		}
+		if err := r.provisioningClusters.Informer().GetIndexer().Resync(); err != nil {
+			logrus.Errorf("error re-syncing provisioningClusters informer in rke2configserver: %v", err)
+		}
+		if err := r.mgmtClusters.Informer().GetIndexer().Resync(); err != nil {
+			logrus.Errorf("error re-syncing mgmtClusters informer in rke2configserver: %v", err)
+		}
 		rw.WriteHeader(http.StatusUnauthorized)
 		return
 	}
+
+	// Short circuit if the path is the connect cluster agent yaml path
+	if ConnectClusterAgentYamlRegex.MatchString(req.URL.Path) {
+		r.connectClusterAgentYAML(rw, req)
+	}
```

Comment:
Should it return here to really short-circuit? 
```suggestion
  return 
	}
```